File: Second Semester(Java and C++)/List 3/C++/Square.cpp
#include "Square.hpp"

#include <stdexcept>

Square::Square(int side, int angle) noexcept(false) {
    if (angle != 90) {
        throw std::invalid_argument("Square's angle must be 90 degrees");
    }

    for (int i = 0; i < 4; i++) sides.push_back(side);
    this->angle = angle;
}

std::string Square::getName() noexcept(true) {
    return "Square";
}

File: Second Semester(Java and C++)/List 3/Java/Task 1/Square.java
public class Square extends Quadrangle {
    public Square(int side, int angle) throws IllegalArgumentException {
        if (angle != 90) {
            throw new IllegalArgumentException("This is not a square!");
        }

        for (int i = 0; i < 4; i++) sides.add(side);
        this.angle = angle;
    }

    @Override
    public String getName() {
        return "Square";
    }
}
